src/main/kotlin/com/cognifide/gradle/aem/pkg/PackagePlugin.kt

```diff
@@ -94,16 +98,19 @@ class PackagePlugin : CommonDefaultPlugin() {
         }
 
         register<PackageDelete>(PackageDelete.NAME) {
-            dependsOn(compose)
+            dependsOn(compose, validate)
             mustRunAfter(deploy, upload, install, activate, uninstall)
         }
         register<PackagePurge>(PackagePurge.NAME) {
-            dependsOn(compose)
+            dependsOn(compose, validate)
             mustRunAfter(deploy, upload, install, activate, uninstall)
         }
         named<Task>(LifecycleBasePlugin.ASSEMBLE_TASK_NAME) {
             dependsOn(compose)
         }
+        named<Task>(LifecycleBasePlugin.CHECK_TASK_NAME) {
+            dependsOn(validate)
```

Comment:
Does compose no longer depend on `check`? Otherwise it’s still a cycle, no?

Reply:
compose does not depend on `check` as I checked yesterday.

Reply:
![image](https://user-images.githubusercontent.com/10673561/77317134-ec502f80-6d0a-11ea-9cec-00d89bab1d4b.png)

I am still thinking about it. It is nice that when building package unit tests are run, but it introduces a problem in assembly packages when merging - circular task dependencies.

Now I am thinking more about running all unit tests separately and pinning it to `develop` task here by simply listing desired test task paths to be run each code change.

Reply:
see https://github.com/Cognifide/gradle-aem-plugin/blob/master/src/main/kotlin/com/cognifide/gradle/aem/pkg/PackagePlugin.kt#L61

when single project has both package and bundle plugin, then `packageCompose` will depend on `test` (not `check` directly)

Reply:
ok tuned it up ;)

![image](https://user-images.githubusercontent.com/10673561/77318992-54544500-6d0e-11ea-9284-beff938df1bb.png)

extended `nestPackageProject` and `mergePackageProject` and `installBundleProject` so that assembly package are running unit tests and package validations of subproject being merged or nested.

Reply:
> compose does not depend on `check` as I checked yesterday.

That must have been an older version of GAP that did this. Sorry…